### ISSUE TITLE
Can't detect custom objects

### ISSUE THREAD

**Issue by martisaju** [AlexeyAB/darknet#2544]
Hello!
I am trying to detect my custom objects but my YOLO network traning process takes a lot of time. I am doing it with Google Colab and it loses its connection before the convergence of the loss function (after 6000 iterations) so it can 't detect any object...

Do you know how can I solve that problem?

*  Also I have a doubt that maybe that's the reason of my problem: When I upload the training images with their corresponding bounding box I don't use entire images, I just upload images of the object I want to detect (with their bounding box):

i.e.
![captura](https://user-images.githubusercontent.com/29142274/53886538-853cdb80-4020-11e9-9d95-9a465df13891.JPG)

instead of:
![30](https://user-images.githubusercontent.com/29142274/53886465-62aac280-4020-11e9-8784-dbceefdb556c.jpg)

Could it be the reason of my problem?

Thank you so much

**Comment by PeterQuinn925** [comment]
Yes. The second one (with a bounding box) is what you want to use.

**Comment by AlexeyAB** [comment]
@martisaju Hi,

Your relative object size should be the same on the training and detection images.
So use the 2nd image for training and detection.

**Comment by middlto** [comment]
@AlexeyAB Hi,

Thank you for your darknet, and I'm trying to use it to detect my custom objects. But I meet the same problem with martisaju. My model can't detect my custom object after trained.

I set classes=1, filters=18 in all three places in my `.cfg` file, and I use 474 pictures with `.txt` files to train my model.

Finally, the iteration is 740 and the avg_loss is 0.851533.

Then I use pictures in my training set to test the model by map command. The average precision is 0.

I don’t know where the problem is. Do you know how can I solve this problem? Thanks a lot!

**Comment by AlexeyAB** [comment]
@middlto 

You should train more than 4000 iterations. You should use GPU for this.

https://github.com/AlexeyAB/darknet#when-should-i-stop-training

> Usually sufficient 2000 iterations for each class(object), but not less than 4000 iterations in total.

**Comment by middlto** [comment]
@AlexeyAB 

Thank you! I will try more iterations.

**Comment by AlexeyAB** [comment]
@middlto Do you use nVidia GPU for training?

**Comment by middlto** [comment]
@AlexeyAB 

Yes, I use nvidia gtx 1080ti for training.

Thank you for your guidance! I trained my model for 2220 iterations, and now its avg_precision is  90.38%. I will save models per 100 epoch, and see which one is better.

**Comment by nancyshaji** [comment]
@martisaju  Is it possible for you to share the training code on Google colab?
I am doing a project where I need to train a custom dataset and I do not have access to a GPU, other than colab

**Comment by martisaju** [comment]
@nancyshaji I didn't keep the training code I used, but I got all the needed information from this site (Part 5 shows how to train on Google colab):
https://github.com/ivangrov/YOLOv3-Series
:)
